src/enums/variant-payloads.md

```diff
@@ -9,10 +9,14 @@ You can define richer enums where the variants carry data. You can then use the
 
 <details>
 
-* In the above example, accessing the `char` in `KeyPress`, or `x` and `y` in `Click` only works within a `match` or an `if let` statement.
-* `match` and `if let` inspect a hidden discriminant field in the `enum`.
+* In the above example, accessing the `char` in `KeyPress`, or `x` and `y` in `Click` only works within a `match` statement.  
+  * In a match, we check some expressions against some possible options, called match arms. Each option is declared with a pattern, an arrow, some code and then a comma. When it's a multi-line block we use a pair curly braces.
```

Comment:
Let's simplify this a bit and explain the lack of fall-through (the syntax is explained by the example itself):

```suggestion
  * The expressions is matched against the patterns from top to bottom. There is no fall-through like in C or C++.
```